library/std/src/sync/mpmc/mod.rs

```diff
@@ -1,8 +1,102 @@
-//! Multi-producer multi-consumer channels.
+//! Multi-producer, multi-consumer FIFO queue communication primitives.
+//!
+//! This module provides message-based communication over channels, concretely
+//! defined among three types:
+//!
+//! * [`Sender`]
+//! * [`Receiver`]
+//!
+//! A [`Sender`] is used to send data to a set of [`Receiver`]s. Both
+//! sender and receiver are clone-able (multi-producer) such that many threads can send
+//! simultaneously to receivers (multi-consumer).
+//!
+//! These channels come in two flavors:
+//!
+//! 1. `channel` function: Creates a channel of unbounded capacity.
+//!
+//! 2. `sync_channel` function: Creates a channel of bounded capacity.
```

Comment:
Probably copy some of the more detailed docs here from mpsc. Also add doc links to channel / sync_channel